test/server/cards/01_SOR/LeiaOrganaAllianceGeneral.spec.js

```diff
@@ -0,0 +1,129 @@
+describe('Leia Organa, Alliance General', function() {
+    integration(function() {
+        describe('Leia\'s undeployed ability', function() {
+            beforeEach(function () {
+                this.setupTest({
+                    phase: 'action',
+                    player1: {
+                        groundArena: ['atst', 'battlefield-marine', 'fleet-lieutenant', { card: 'rebel-pathfinder', exhausted: true }],
+                        spaceArena: ['tieln-fighter', 'alliance-xwing'],
+                        leader: 'leia-organa#alliance-general'
+                    },
+                    player2: {
+                        groundArena: ['regional-governor'],
+                        spaceArena: ['tie-advanced']
+                    }
+                });
+            });
+
+            it('should attack with two Rebel units', function () {
+                this.player1.clickCard(this.leiaOrgana);
+                this.player1.clickPrompt('Attack with a Rebel unit');
+                expect(this.player1).toBeAbleToSelectExactly([this.battlefieldMarine, this.fleetLieutenant, this.allianceXwing]);
+                expect(this.player1).not.toHaveEnabledPromptButton('Pass ability');
+
+                this.player1.clickCard(this.battlefieldMarine);
+                this.player1.clickCard(this.regionalGovernor);
+                expect(this.battlefieldMarine.exhausted).toBe(true);
+                expect(this.regionalGovernor.damage).toBe(3);
+                expect(this.battlefieldMarine.damage).toBe(1);
+
+                // second attack
+                expect(this.player1).toBeActivePlayer();
+                expect(this.player1).toBeAbleToSelectExactly([this.fleetLieutenant, this.allianceXwing]);
+                expect(this.player1).toHaveEnabledPromptButton('Pass ability');
+                this.player1.clickCard(this.allianceXwing);
+                expect(this.player1).toHaveEnabledPromptButton('Pass ability');
```

Comment:
Should the pass ability be present once the card is selected here?

Reply:
I would argue yes, since the cost (exhausting the attacker) hasn't been paid yet so there's no issue if the user changes their mind and decides to pass at the second step. We could talk to Ross, but it would actually require extra work to keep it from being present at that stage - there's a check in the target resolver to see if passing is allowed and if so it just presents the button, it doesn't have awareness of whether the button was presented in an earlier stage already.